### ISSUE TITLE
Few ideas

### ISSUE THREAD

**Issue by johannesdedoper** [jeanslack/Videomass#87]
Hello,

I got another few ideas ;)
1. sorting of the input files on filename/duration/mediatype or size
2. counter in screen input files 
3. copy preset
![sortbyclicking](https://user-images.githubusercontent.com/100475891/158781384-78b8be9d-69be-49c3-8caa-4510ce846a06.png)
![copypreset](https://user-images.githubusercontent.com/100475891/158781389-b062f9bc-5653-47cc-bcec-14947a06943b.png)

cheers,
J.

**Comment by jeanslack** [comment]
Hello @johannesdedoper

I am evaluating your ideas but could you be more specific in your requests please?

> 1. sorting of the input files on filename/duration/mediatype or size

IMHO one or the other thing, in this case I think it makes more sense to sort by `Media Type`

> 3. copy preset

What exactly should be copied? Could you give me an example on a real use case?

**Comment by johannesdedoper** [comment]
Hello @jeanslack,
thnx for your reply.

Regarding my points:
1. sorting columns
Exactly, I want to sort based on "filename" OR "duration" OR "mediatype" OR "size". 
Why would I want that you may ask? well the reason is that I want to convert files is when it results in a smaller size file with a acceptable quality. If files are already compressed enough it will not be necessary to encode them.
What I do is sort all files on my disk based on size (MB, GB..whatever). 
Then I drop all files eg. bigger than 2 GB into Videomass, 
Then I choose the files that can be compressed. My criteria is based on a duration/size ratio. 
Personally I always aim for a quality that will approx. lead to a filesize of 1GB in h265 for 1 hour film/video. So for a 2 hour movie I aim for a 2GB h265 file. So I calculate the duration/size ratio: 2000MB/120minutes = 16,7 MB/minute

add column RATIO?
Of course in a ideal situation there would be an additional column with a ratio (filesize in MB divided by duration in minutes). 
Than it would be very easy for me to sort the input files based on this ratio. This way I can easily decide what to encode and remove the files from the input file list that are already compressed enough. 
That would be really magnificent and probably useful for a lot of people ;)   

Speaking of removing input files:
it would be nice to have the possibility to select several input files in this queue list with holding the SHIFT or CTRL key and than click on "REMOVE" (or just push the delete button on your keyboard).

3. copy preset
What I like to do is use a existing preset eg. "01. h265_nvenc 2600kb audio aac" and just adjust a few things inclusing the name and then have a new preset. So in this case I was thinking: 
I select the preset (in the above picture it's highlighted in green)  
Than I click on the "copy" (or duplicate) button and the selected preset "01. h265_nvenc 2600kb audio aac" is copied and placed beneath (or above, whatever) the original and shows up with a non- conflicting name like "01. h265_nvenc 2600kb audio aac - copy"
Than I can click on the "01. h265_nvenc 2600kb audio aac - copy" and adjust the name and parameters. 
The reason for this is that you will not need to type in all of the parameters again but slighty adjust them to get a new preset.

Hope this clarifies these ideas?
If not please let me know.
Cheers, J

**Comment by jeanslack** [comment]
Thanks for the details, it's all a lot clearer now.
I don't know if it will be possible to satisfy all your requests (ideas) but I will definitely take them into consideration for testing.
I will let you know.

Good day !

**Comment by johannesdedoper** [comment]
Thnx for considering!
Cheers, Ciao
J.

**Comment by johannesdedoper** [comment]
sorting :+1:  Nice

**Comment by jeanslack** [comment]
Hello @johannesdedoper,

Hell, I totally forgot there was an option to click on the column headers to change the order of the items, that's why I didn't understand your screenshot at first. Despite this it seems to me that I have implemented most of your ideas.
Soon I will merge all PRs with the master and post a new version. 
Thnx for the inputs!

**Comment by johannesdedoper** [comment]
Thank YOU for implementing :+1:

**Comment by johannesdedoper** [comment]
I tested the 3.5.8 source and I'm very happy with the changes. 
Even tried the concatenate with pictures to video. Works fine, of course the pictures need to have the same resolution. 
So the duplication of presets works fine (saves a lot of time fine tuning and testing presets)!!
The numbering is nice so I can easily check when encoding big batches if all files are in the list.
The sorting seems to be working but I noticed it only sorts ascending. So when you click twice on the same variable (eg. duration) nothing happens the second time/click. Normally first click sorts ascending, second descending. 
At first a bit confusing as most times it works as described above. 

Still thnx again!!
Cheers,
J.

**Comment by jeanslack** [comment]
Also you can select multiple lines now (using CTRL or SHIFT key) and perform multiple deletions.

> The sorting seems to be working but I noticed it only sorts ascending. So when you click twice on the same variable (eg. duration) nothing happens the second time/click. Normally first click sorts ascending, second descending.
At first a bit confusing as most times it works as described above.

Oh, sorry, I forgot to point this out: for descending sorting you should use right mouse click. You can also do this the first time (not only after having already clicked for ascending sorting). But if you also believe that this is unusual, let's change it. 
Let me know.

**Comment by johannesdedoper** [comment]
> Also you can select multiple lines now (using CTRL or SHIFT key) and perform multiple deletions.

YESS! great. Almost  forgot about that.

> Oh, sorry, I forgot to point this out: for descending sorting you should use right mouse click. You can also do this the first time (not only after having already clicked for ascending sorting). But if you also believe that this is unusual, let's change it.

Well for me clicking goes from ascending to descending and back to ascending etc. In my file explorer, video editing software, emailclient (Thunderbird) it works that way, so I´m used to that. I think it's more common?
But it's not really a issue as I know now how it works.
Thnx again, for all the nice updates!!!!
Cheers,
J.

**Comment by jeanslack** [comment]
> Well for me clicking goes from ascending to descending and back to ascending etc. In my file explorer, video editing software, emailclient (Thunderbird) it works that way, so I´m used to that. I think it's more common?

Yes it is.
OK, switched to common way (from ascending to descending and back to ascending with left mouse click only)

**Comment by johannesdedoper** [comment]
Nice! great. thnx AGAIN ;)
